src/main/java/no/uio/ifi/localega/doa/rest/StreamingController.java

```diff
@@ -63,26 +60,23 @@
     @Value("${crypt4gh.private-key-password-path}")
     private String crypt4ghPrivateKeyPasswordPath;
 
+    @SuppressWarnings("unchecked")
     @GetMapping("/{fileId}")
-    public ResponseEntity<?> files(@RequestHeader(value = "Authorization") String token,
-                                   @RequestHeader(value = "Public-Key", required = false) String publicKey,
+    public ResponseEntity<?> files(@RequestHeader(value = "Public-Key", required = false) String publicKey,
                                    @PathVariable(value = "fileId") String fileId,
                                    @RequestParam(value = "destinationFormat", required = false) String destinationFormat,
                                    @RequestParam(value = "startCoordinate", required = false) String startCoordinate,
                                    @RequestParam(value = "endCoordinate", required = false) String endCoordinate) throws Exception {
-        DecodedJWT decodedJWT = jwtVerifier.verify(token.replace("Bearer ", ""));
-        String subject = decodedJWT.getSubject();
-        log.info("User {} is authenticated and is attempting to download the file: {}", subject, fileId);
-        Set<String> datasetIds = Arrays.stream(decodedJWT.getClaim("authorities").asArray(String.class)).collect(Collectors.toSet());
+        Set<String> datasetIds = (Set<String>) request.getAttribute(AAIAspect.DATASETS);
         if (!checkPermissions(fileId, datasetIds)) {
-            log.info("User doesn't have permissions to access this file, abort");
+            log.info("User doesn't have permissions to download requested file: {}", fileId);
             return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
-        log.info("User has permissions to access the file, initializing download");
+        log.info("User has permissions to download requested file: {}", fileId);
```

Comment:
we need to be able to identify which user: we had a whole thing about it in `ega-data-api` https://github.com/neicnordic/ega-data-api/pull/13

One can used the `sub` claim from the JWT payload to identify the `user` by his/her elixir-id it will take the form or `<id>@elixir-europe.org`

The same goes for all the logs where the user is referenced.

Reply:
This is logged in the `AAIAspect: https://github.com/neicnordic/LocalEGA-DOA/pull/13/files#diff-69b00d226347e9bd8a77f6d23454d5f2R62`

Reply:
would it be complicated to add here as well ? in case there are many users, it will be easier to track

Reply:
So, this aspect wraps every REST method of the DOA (2 methods in `MetadataController` and 1 in `StreamingController`), so there's no need to log it explicitly within those methods (`AAIAspect` acts as a decorator). It was actually the reason I did it this way: didn't want to duplicate auth and logging code in 3 places. Or do I miss the point?